src/ert/shared/models/ensemble_smoother.py

```diff
@@ -24,6 +34,178 @@ def setAnalysisModule(self, module_name: str) -> None:
+
+        await self._run_hook(
+            HookRuntime.PRE_UPDATE, prior_context.iteration, loop, threadpool
+        )
+
+        # Send ANALYSIS_STARTED
+        await self.dispatch(
+            CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_ANALYSIS_STARTED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        experiment_logger.debug("running update...")
+        try:
+            await loop.run_in_executor(
+                threadpool, self.facade.smoother_update, prior_context
+            )
+        except ErtAnalysisError as e:
+            experiment_logger.exception("analysis failed")
+            # Send EXPERIMENT_FAILED
+            await self.dispatch(
+                CloudEvent(
+                    {
+                        "type": identifiers.EVTYPE_EXPERIMENT_FAILED,
+                        "source": f"/ert/experiment/{self.id_}",
+                        "id": str(uuid.uuid1()),
+                    },
+                    {
+                        "error": str(e),
+                    },
+                ),
+                prior_context.iteration,
+            )
+            raise ErtRunError(
+                f"Analysis of simulation failed with the following error: {e}"
+            ) from e
+
+        experiment_logger.debug("update complete")
+
+        # Send ANALYSIS_ENDED
+        await self.dispatch(
+            CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_ANALYSIS_ENDED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        await self._run_hook(
+            HookRuntime.POST_UPDATE, prior_context.iteration, loop, threadpool
+        )
+
+        # Create an update object in storage
+        analysis_module_name = self.ert().analysisConfig().activeModuleName()
+        update_id = await loop.run_in_executor(
+            threadpool, self._post_update_data, ensemble_id, analysis_module_name
+        )
+
+        self.ert().getEnkfFsManager().switchFileSystem(prior_context.target_fs)
+
+        experiment_logger.debug("create context for iter 1")
+        rerun_context = await loop.run_in_executor(
+            threadpool, partial(self.create_context, prior_context=prior_context)
+        )
+
+        await loop.run_in_executor(
+            threadpool,
+            self.ert().createRunPath,
+            rerun_context,
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_SIMULATION, rerun_context.iteration, loop, threadpool
+        )
+
+        # Push ensemble, parameters, observations to new storage
+        ensemble_id = await loop.run_in_executor(
+            threadpool, self._post_ensemble_data, update_id
```

Comment:
`_post_ensemble_data` calls `self.setPhaseName()` which should maybe invoke a call to the experiment server?

Reply:
That will happen later when the experiment server supports it.